### PR TITLE
[DI] Create `ADR` search result item extension

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Migrate the `AdrSearchResultItemExtension` to the declarative integration format.

### Known limitations:

- It doesn't support icon customization for now.

### Usage example:

> [!Note]
> Currently, it is in an experimental stage and is not recommended for production

For frontend applications that support declarative integration, it is only necessary to install the following packages:

```sh
# install the catalog, search, and adr plugins packages
yarn add @backstage/plugin-catalog @backstage/plugin-search @backstage/plugin-adr
```

Optionally you can customize the search results via `app-config.yaml` file:

```yaml
# app-config.yaml
# example disabling analytics tracking and setting a line clamp
app:
  experimental:
    packages: 'all' # ✨

  extensions:
    - plugin.search.result.item.adr:
        config:
          noTrack: true
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/adr/src/alpha.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * React extension used to render results on Search page or modal
+ * @alpha
+ */
+export const AdrSearchResultListItemExtension =
+  createSearchResultListItemExtension({
+    id: 'adr',
+    at: 'plugin.search.page/items',
+    configSchema: createSchemaFromZod(z =>
+      z.object({
+        // TODO: Define how the icon can be configurable
```
Comment (camilaibs) [inline plugins/adr/src/alpha.tsx:41]: Do you happen to have any ideas, @Rugvip?

Comment (Rugvip) [reply plugins/adr/src/alpha.tsx:38]: Could be an input, could be an API, could be something else. I think it's fine to leave for now :p

**Review by Rugvip — APPROVED** [review]
Sweet! :+1:

**Comment by camilaibs** [comment]
Hi @kuangp, since this is experimental, we'll merge it, but if you have any questions, don't hesitate to ping us 🙂

**Pull request merged**